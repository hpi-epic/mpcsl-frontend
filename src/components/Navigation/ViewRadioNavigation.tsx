import { Radio } from 'antd';
import { RadioChangeEvent } from 'antd/lib/radio';
import React from 'react';
import { Routes } from '../../types';

const RadioButton = Radio.Button;
const RadioGroup = Radio.Group;

interface IPropsRadioNavigation {
  value: Routes;
  onChange: (e: RadioChangeEvent) => void;
}

function ViewRadioNavigation(props: IPropsRadioNavigation) {
  return (
    <RadioGroup
      buttonStyle='solid'
      value={props.value}
      onChange={props.onChange}
      size='small'
    >
      <RadioButton value={Routes.observationMatricesManager}>
        Observation Matrices Management
      </RadioButton>
      <RadioButton value={Routes.experimentManager}>
        Experience Management
      </RadioButton>
    </RadioGroup>
  );
}

export default ViewRadioNavigation;
