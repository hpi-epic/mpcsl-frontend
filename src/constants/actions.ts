export const ADD_GRAPH = 'ADD_GRAPH';
export type ADD_GRAPH = 'ADD_GRAPH';
export const ADD_NODE = 'ADD_NODE';
export type ADD_NODE = 'ADD_NODE';
export const NEW_GRAPH_LAYOUT = 'NEW_GRAPH_LAYOUT';
export type NEW_GRAPH_LAYOUT = 'NEW_GRAPH_LAYOUT';
export const TOGGLE_FREEZE_LAYOUT = 'TOGGLE_FREEZE_LAYOUT';
export type TOGGLE_FREEZE_LAYOUT = 'TOGGLE_FREEZE_LAYOUT';
