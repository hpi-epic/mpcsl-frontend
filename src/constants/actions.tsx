export const ADD_GRAPH = 'ADD_GRAPH';
export type ADD_GRAPH = typeof ADD_GRAPH;
