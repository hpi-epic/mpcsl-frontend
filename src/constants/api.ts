const baseApiUrl = '/api';

const Endpoints = {
  observationMatrix: baseApiUrl + '/dataset',
  observationMatrices: baseApiUrl + '/datasets',
  experiment: baseApiUrl + '/experiment',
  allExperiments: baseApiUrl + '/experiments',
  job: baseApiUrl + '/job',
  allJobs: '/jobs',
  allResults: baseApiUrl + '/results',
  result: baseApiUrl + '/result',
  nodeDistribution: (nodeID: string) => `${baseApiUrl}/node/${nodeID}/marginal`,
};

export default Endpoints;
