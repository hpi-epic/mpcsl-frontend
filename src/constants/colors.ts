const colors = {
  contentBackground: '#eff1ef'
};

export default colors;
