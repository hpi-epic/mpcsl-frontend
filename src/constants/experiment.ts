export enum IndepenceTests {
  gaussCI = 'gaussCI',
  disCI = 'disCI',
  binCI = 'binCI',
}
