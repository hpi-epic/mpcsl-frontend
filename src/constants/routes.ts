export enum Routes {
  manager = '/manager',
  observationMatricesManager = '/manager/observationMatrices',
  experimentManager = '/manager/experiments',
  runtimeManager = '/runtime',
  graphExplorer = '/graph-explorer',
  graphExplorerSelection = '/graph-explorer/selection',
  graphExplorerAnnotate = '/graph-explorer/annotate',
  graphExplorerCausalExploration = '/graph-explorer/causal-exploration',
  experimentDetails = '/manager/experimentdetails',
}
