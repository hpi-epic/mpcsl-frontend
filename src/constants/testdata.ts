export const graph = {
  nodes: [
    { id: 0, label: 'a' },
    { id: 1, label: 'b' },
    { id: 2, label: 'c' },
    { id: 3, label: 'd' },
    { id: 4, label: 'e' },
    { id: 5, label: 'f' },
    { id: 6, label: 'g' },
    { id: 7, label: 'h' },
    { id: 8, label: 'i' },
    { id: 9, label: 'j' },
  ],
  links: [
    { source: 1, target: 2 },
    { source: 1, target: 4 },
    { source: 1, target: 8 },
    { source: 2, target: 4 },
    { source: 2, target: 9 },
    { source: 3, target: 6 },
    { source: 3, target: 8 },
    { source: 4, target: 5 },
    { source: 5, target: 8 },
    { source: 7, target: 8 },
  ],
};
