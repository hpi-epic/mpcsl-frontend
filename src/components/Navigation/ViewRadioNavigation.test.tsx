import React from 'react';
import renderer from 'react-test-renderer';
import ViewRadioNavigation from './ViewRadioNavigation';
import { Routes } from '../../constants/routes';

const routes = [Routes.observationMatricesManager, Routes.experimentManager];

describe('<ViewRadioNavigation />', () => {
  routes.forEach(route => {
    it(`renders correctly for route ${route}`, () => {
      const tree = renderer
        .create(
          <ViewRadioNavigation value={route} onChange={() => undefined} />
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
