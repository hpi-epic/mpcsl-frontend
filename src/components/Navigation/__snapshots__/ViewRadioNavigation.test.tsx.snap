// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewRadioNavigation /> renders correctly for route /manager/experiments 1`] = `
<div
  className="ant-radio-group ant-radio-group-solid ant-radio-group-small"
>
  <label
    className="ant-radio-button-wrapper"
  >
    <span
      className="ant-radio-button"
      style={Object {}}
    >
      <input
        checked={false}
        className="ant-radio-button-input"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="radio"
        value="/manager/observationMatrices"
      />
      <span
        className="ant-radio-button-inner"
      />
    </span>
    <span>
      Observation Matrices Management
    </span>
  </label>
  <label
    className="ant-radio-button-wrapper ant-radio-button-wrapper-checked"
  >
    <span
      className="ant-radio-button ant-radio-button-checked"
      style={Object {}}
    >
      <input
        checked={true}
        className="ant-radio-button-input"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="radio"
        value="/manager/experiments"
      />
      <span
        className="ant-radio-button-inner"
      />
    </span>
    <span>
      Experiment Management
    </span>
  </label>
</div>
`;

exports[`<ViewRadioNavigation /> renders correctly for route /manager/observationMatrices 1`] = `
<div
  className="ant-radio-group ant-radio-group-solid ant-radio-group-small"
>
  <label
    className="ant-radio-button-wrapper ant-radio-button-wrapper-checked"
  >
    <span
      className="ant-radio-button ant-radio-button-checked"
      style={Object {}}
    >
      <input
        checked={true}
        className="ant-radio-button-input"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="radio"
        value="/manager/observationMatrices"
      />
      <span
        className="ant-radio-button-inner"
      />
    </span>
    <span>
      Observation Matrices Management
    </span>
  </label>
  <label
    className="ant-radio-button-wrapper"
  >
    <span
      className="ant-radio-button"
      style={Object {}}
    >
      <input
        checked={false}
        className="ant-radio-button-input"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="radio"
        value="/manager/experiments"
      />
      <span
        className="ant-radio-button-inner"
      />
    </span>
    <span>
      Experiment Management
    </span>
  </label>
</div>
`;
